libs/product-composite/routing/src/effects/product-page.effects.ts

```diff
@@ -39,39 +41,51 @@ import {
 } from '../config/public_api';
 
 /**
- * IE compatible query params getter.
+ * Builds the apply actions from the list of selected options.
  */
-// TODO: replace with URLSearchParams when IE support is dropped
-const getQueryParams = (path: string): Record<string, string> => {
-  const params = {};
-  path.replace(
-    /[?&]+([^=&]+)=([^&]*)/gi,
-    (_, key, value) => params[key] = decodeURIComponent(value),
-  );
-  return params;
-};
+function buildApplyActions<T extends DaffCompositeProduct = DaffCompositeProduct>(product: DaffCompositeProduct, itemId: DaffCompositeProductItem['id'], selectedOptions: DaffCompositeProductItemOption['id'][]): DaffCompositeProductApplyOption<T>[] {
```

Comment:
Can this function be a lambda? Can you write a test covering what happens when the incoming URL is a legitimate base64 object but it doesn't match one of the items/selections?

Reply:
it can be but its more performant and expressive as it is, IMO. I've added that test case